build.js

```diff
@@ -1,88 +1,271 @@
 #!/usr/bin/env node
 
 import child_process from 'child_process';
+import fs from 'fs';
+import os from 'os';
 import path from 'path';
 import process from 'process';
 
+import { getFiles, getTestFiles, all_subdirs } from './files.js';
+
+const production = process.env.NODE_ENV === 'production';
+const useWasm = os.arch() != 'x64';
+
 // ensure node_modules is present and up to date
 child_process.spawnSync('tools/node-modules', ['make_package_lock_json'], { stdio: 'inherit' });
 
+// List of directories to use when resolving import statements
+const nodePaths = ['pkg/lib'];
+
+// context options for distributed pages in dist/
+const pkgOptions = {
+    ...!production ? { sourcemap: "external" } : {},
+    bundle: true,
+    external: ['*.woff', '*.woff2', '*.jpg', '*.svg', '../../assets*'], // Allow external font files which live in ../../static/fonts
+    legalComments: 'external', // Move all legal comments to a .LEGAL.txt file
+    loader: {
+        ".js": "jsx",
+        ".py": "text",
+        ".sh": "text",
+    },
+    minify: production,
+    nodePaths,
+    outbase: './pkg',
+    outdir: "./dist",
+    target: ['es2020'],
+};
+
+// context options for qunit tests in qunit/
+const qunitOptions = {
+    bundle: true,
+    // show "build started/finished" messages in watch mode
+    logLevel: 'info',
+    minify: false,
+    nodePaths,
+    outbase: './pkg',
+    outdir: "./qunit",
+    loader: {
+        ".sh": "text",
+    },
+};
+
 const parser = (await import('argparse')).default.ArgumentParser();
 parser.add_argument('-c', '--config', { help: "Path to webpack.config.js", default: "webpack.config.js" });
 parser.add_argument('-r', '--rsync', { help: "rsync bundles to ssh target after build", metavar: "HOST" });
 parser.add_argument('-w', '--watch', { action: 'store_true', help: "Enable watch mode" });
-parser.add_argument('-e', '--no-eslint', { action: 'store_true', help: "Disable eslint linting" });
-parser.add_argument('-s', '--no-stylelint', { action: 'store_true', help: "Disable stylelint linting" });
+parser.add_argument('-e', '--no-eslint', { action: 'store_true', help: "Disable eslint linting", default: production });
```

Comment:
So when it's a production build the lint options are by default disabled now. So in CI we rely on 'test/static-code' for linting? This latter does not have stylelint.

Reply:
Right - that run for the devel package which is enough.